docs/tutorial_deps.rst

```diff
@@ -2,67 +2,86 @@
 This is achieved with the following specially decorated function:
 
 .. literalinclude:: ../tutorials/deps/osu_benchmarks.py
-   :lines: 37-43
+   :lines: 104-109
 
 The :func:`@require_deps <reframe.core.decorators.require_deps>` decorator will bind the arguments passed to the decorated function to the result of the dependency that each argument names.
-In this case, it binds the ``OSUBuildTest`` function argument to the result of a dependency named ``OSUBuildTest``.
+In this case, it binds the ``OSUDownloadTest`` function argument to the result of a dependency named ``OSUDownloadTest``.
 In order for the binding to work correctly the function arguments must be named after the target dependencies.
 However, referring to a dependency only by the test's name is not enough, since a test might be associated with multiple programming environments.
 For this reason, a dependency argument is actually bound to a function that accepts as argument the name of a target programming environment.
-If no arguments are passed to that function, as in this example, the current programming environment is implied, such that ``OSUBuildTest()`` is equivalent to ``OSUBuildTest(self.current_environ.name)``.
+If no arguments are passed to that function, as in this example, the current programming environment is implied, such that ``OSUDownloadTest()`` is equivalent to ``OSUDownloadTest(self.current_environ.name, self.current_partition.name)``.
 This call returns the actual test case of the dependency that has been executed.
-This allows you to access any attribute from the target test, as we do in this example by accessing the target test's stage directory, which we use to construct the path of the executable.
+This allows you to access any attribute from the target test, as we do in this example by accessing the target test's stage directory, which we use to construct the sourcesdir of the test.
+
+For the next test we need to use the OSU benchmark binaries that we just built, so as to run the MPI ping-pong benchmark.
+Here is the relevant part:
+
+.. literalinclude:: ../tutorials/deps/osu_benchmarks.py
+   :lines: 13-44
+
+First, since we will have multiple similar benchmarks, we move all the common functionality to the :class:`OSUBenchmarkTestBase` base class.
+Again nothing new here; we are going to use two nodes for the benchmark and we set :attr:`sourcesdir <reframe.core.pipeline.RegressionTest.sourcesdir>` to ``None``, since none of the benchmark tests will use any additional resources.
+Similar to before, we will define the dependencies with the the following line:
+
+.. literalinclude:: ../tutorials/deps/osu_benchmarks.py
+   :lines: 32
+
+Here we tell ReFrame that this test depends on a test named ``OSUBuildTest`` ``by_env``.
+This means that it will depend only on the instances of this test that have the same environment.
+The partition will be different in this case.
```

Comment:
This and the sentence above are a bit hard to follow. Can you explain what you want to say?

Reply:
I just wanted to briefly explain what `by_env` means. They will have same environment but different partition.